src/compas/scene/sceneobject.py

```diff
@@ -22,7 +21,46 @@
 from .descriptors.protocol import DescriptorProtocol
 
 
-class SceneObject(TreeNode):
+def SceneObjectFactory(item=None, scene=None, context=None, **kwargs):
```

Comment:
function name breaks naming conventions. why not just snake case?

Reply:
type hint?

Reply:
I forget to switch pattern from class naming